modules/carto/src/style/palette.js

```diff
@@ -0,0 +1,83 @@
+import * as cartoColors from 'cartocolor';
+import {assert} from './utils';
+
+export const DEFAULT_PALETTE = 'PurpOr';
+export const NULL_COLOR = [204, 204, 204];
+export const OTHERS_COLOR = [119, 119, 119];
+
+export default function getPalette(name, numCategories) {
+  const palette = cartoColors[name];
+  let paletteIndex = numCategories;
+
+  assert(palette, `Palette "${name}" not found. Expected a CARTOColors string`);
+
+  const palettesColorVariants = Object.keys(palette)
+    .filter(p => p !== 'tags')
+    .map(Number);
+
+  const longestPaletteIndex = Math.max(...palettesColorVariants);
+  const smallestPaletteIndex = Math.min(...palettesColorVariants);
+
+  if (!Number.isInteger(numCategories) || numCategories > longestPaletteIndex) {
+    paletteIndex = longestPaletteIndex;
+  } else if (numCategories < smallestPaletteIndex) {
+    paletteIndex = smallestPaletteIndex;
+  }
+
+  let colors = palette[paletteIndex];
```

Comment:
This could cause a bug because the palette is being updated (color.slice). Consider cloning the object `[ ...palette[paletteIndex] ]`.

Reply:
:wave:
`getPalette` is called each time we need an update, so there is no problem as long as `colors` is being created again